plugins/BEdita/API/src/Controller/UsersController.php

```diff
@@ -153,7 +157,9 @@ public function delete($id)
     {
         $this->request->allowMethod('delete');
 
-        $user = $this->Users->get($id);
+        $user = $this->Users->get($id, [
+            'conditions' => ['deleted' => 0]
+        ]);
         if (!$this->Users->delete($user)) {
```

Comment:
Right.
Solved and modified unit test.
I had to use `ObjectsTable` instead of `UserTable` in `UsersController::delete()` - I'm opening a new issue for that